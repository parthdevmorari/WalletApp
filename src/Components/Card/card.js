import { View, Text, Image } from 'react-native'
import React from 'react'
import styles from '../Card/cardStyle';
import {ShowhideCard} from '@Component'
import * as resources from 'resources';

const Card = (props) => {
    const [isVisible,setIsVisible]= React.useState(true)
	const { cardNo,title,cardDate,cardCvv,imageSource, imageSource2} = props;

    return (
        <View style={[styles.container]}>
            <ShowhideCard 
			title = {'Hide card number'}
			imageSource = {isVisible?resources.show:resources.hide}
            onPress={()=>{setIsVisible(!isVisible)}}
            style={styles.showhideButton}
			/>
       
        <View style={[styles.cardContainer]}>
                <View style = {styles.cardLogoContainer}>
                <Image source={imageSource} style = {styles.LogoStyle}  />
                </View>
                <View style = {styles.detailStyle}>
            <Text style={[styles.nameText]}>{title}</Text>
          { isVisible? <Text style={[styles.NumberText]}>{cardNo}</Text>:
          <Text style={[styles.NumberText2]}>{".... .... .... ...."}</Text>
          }
            <View style={[styles.txtContainer]}>
                <Text style={[styles.dateText]}>{cardDate}</Text>
                <Text style={[styles.cvvText]}>{"CVV:"}</Text>
                { isVisible? 
                <Text style={[styles.cvvText]}>{cardCvv}</Text> :
              <Text style={[styles.cvvText]}>{"* * *"}</Text>
          }
            </View>
            </View>
            <View style = {styles.cardLogoContainer2}>
                <Image source={imageSource2} style = {styles.LogoStyle2}  />
                </View>
        </View>
        </View>
    )
}

export default Card