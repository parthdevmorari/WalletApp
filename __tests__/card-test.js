import 'react-native';
import React from 'react';
import card from '../src/Components/Card/card'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('Card Component', () => {
  test('Card Component', () => {
    const wrapper = shallow(<card />);
    expect(wrapper.exists()).toBe(true);
  });
  it('Card  toMatchSnapshot', () => {
    const wrapper = shallow(<card />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});