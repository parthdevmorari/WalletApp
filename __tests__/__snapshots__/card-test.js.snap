// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Card  toMatchSnapshot 1`] = `<card />`;
